department-of-veterans-affairs/va.gov-cms
# VACMS-12568: Update smart date and fix errors

## Description
This PR:

- updates smart_date to the latest stable version: 3.7.1
- corrects errors encountered when attempting to update previously
- Updates/removes patches as necessary
- Removes a "hotfix" update that was causing an error when attempting to update recurring instances. This was replaced with a formal d.o contributed patch with new unit tests.
- And more! See the [issue tasks](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12568#:~:text=a%20single%20PR.-,Tasks,-The%20extent%20of) for a complete list.

Relates to #12568
Closes #10490

## Screenshots
Event Setup:
![Screenshot 2023-03-02 at 9 59 42 AM](https://user-images.githubusercontent.com/221539/222512801-20188eb2-84ba-4462-9293-814ced5349e2.png)

## QA steps
As a content admin
1. Create a new Event at /node/add/event. Configure the event as shown in the 'Event Setup' screenshot above.
2. Save the Event
   - [x] Validate that the start time for all events in MST begin at 10:00 am. This confirms the former Phoenix timezone issue is resolved.
3. Edit the Event
4. Click the 'Edit event series' button
5. Click the 'Edit' link in the 'Manage event series' modal for any of the events.
6. Change the start time for the event. Click 'Save'
    - [x] Validate you are returned to the 'Manage event series' modal.
7. Close the modal.
8. Save the Event node.
   - [x] Validate the node saved as expected.

### Definition of Done


### Select Team for PR review

### CMS user-facing announcement

Is an announcement needed to let editors know of this change?
